File: Leetcode/algorithms/Java/longest common prefix/longestCommonPrefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0)
            return "";

        String first = strs[0];
        for (int i = first.length(); i >= 0; i--) {
            String prefix = first.substring(0, i);
            boolean allMatch = true;
            for (String word : strs) {
                if (!word.startsWith(prefix)) {
                    allMatch = false;
                    break;
                }
            }
            if (allMatch) {
                return prefix;
            }
        }
        return "";
    }
}

File: Leetcode/algorithms/JS/longest common prefix/longestCommonPrefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if(strs.length==0) return "";
 
    let first=strs?.[0];
    for(let i=first.length; i>=0;i--){
     let prefix= first?.substring(0,i);
     if(strs.every(word=> word.startsWith(prefix)))
         return prefix;
    }
    return ""
 };